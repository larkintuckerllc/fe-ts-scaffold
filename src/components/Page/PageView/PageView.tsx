import { List } from 'immutable';
import React from 'react';
import AnotherCounter from './AnotherCounter';
import Ant from './Ant';
import Async from './Async';
import Counter from './Counter';
import Frame from './Frame';
import InfiniteScroll from './InfiniteScroll';
import Listing from './Listing';
import Media from './Media';
import Paginated from './Paginated';
import Risky from './Risky';
import Stateful from './Stateful';
import Styling from './Styling';
import Wrapped from './Wrapped';

interface PageViewProps {
  adder: List<string>;
  counter: number;
  infinite: boolean;
  add(value: string): void;
  decrement(): void;
  increment(): void;
  toggleInfinite(): void;
}

const PageView = ({
  add,
  adder,
  counter,
  decrement,
  increment,
  infinite,
  toggleInfinite,
}: PageViewProps) => {
  if (infinite) {
    return <InfiniteScroll />;
  }
  return (
    <Frame>
      <Ant />
      <AnotherCounter />
      <Async />
      <Counter counter={counter} decrement={decrement} increment={increment} />
      <button onClick={toggleInfinite}>Show Infinite</button>
      <Listing add={add} items={adder.toJS()} />
      <Media />
      <Paginated />
      <Risky />
      <Stateful />
      <Styling />
      <Wrapped text="test" />
    </Frame>
  );
};
export default PageView;
