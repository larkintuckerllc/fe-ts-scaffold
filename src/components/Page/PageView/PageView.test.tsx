import Enzyme, { shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import { List } from 'immutable';
import React from 'react';
import PageView from './PageView';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

const getDefaultProps = () => ({
  add: jest.fn(),
  adder: List<string>([]),
  counter: 0,
  decrement: jest.fn(),
  increment: jest.fn(),
  infinite: false,
  toggleInfinite: jest.fn(),
});
window.scroll = jest.fn();

describe('PageView component', () => {
  it('shallow renders without crashing', () => {
    const defaultProps = getDefaultProps();
    shallow(<PageView {...defaultProps} />);
  });

  it('renders differently with infinite', () => {
    const { infinite, ...defaultProps } = getDefaultProps();
    const wrapper = shallow(<PageView {...defaultProps} infinite={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
