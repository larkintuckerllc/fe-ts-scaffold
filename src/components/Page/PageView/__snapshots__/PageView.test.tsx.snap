// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageView component renders differently with infinite 1`] = `<Connect(ReactInfiniteScrollController) />`;
